todo_app/app.py

```diff
@@ -1,11 +1,81 @@
-from flask import Flask
+from flask import Flask, request, render_template, redirect, url_for
 
 from todo_app.flask_config import Config
+from todo_app.data.session_items import get_item, get_items, save_item, add_item, delete_item
+from todo_app.utils import SimpleValidation
 
 app = Flask(__name__)
 app.config.from_object(Config())
 
+# Route for the index page
+# Display the to do items on the index page - retrieving them from the session. Sort Alphabetically.
 
 @app.route('/')
 def index():
-    return 'Hello World!'
+    error = request.args.get("error") or False
+    items = get_items()
+
+    items = sorted(items, key=lambda item: item["status"])
+
+    return render_template("index.html", items=items, error=error)
+
+#End of Index
```

Comment:
We'll talk about clean code more in module 2, but I'll comment on your code comments now (to save yourself some time).

Generally speaking we shouldn't need code comments. They're only useful for genuinely surprising or difficult to understand bits of code. What we instead aim for is "self-documenting" code by writing things clearly.

Code comments take time to write, take up space and require maintenance (or more likely end up incorrect and actually cause problems)

E.g. Here, "# Route for the index page" is not more helpful than the fact you have a function called "index" with an "app.route" decorator.
For `Sort Alphabetically` you could instead change your code to `items_sorted_alphabetically = sorted(...etc`

 And "# End of Index" isn't necessary for a function containing only four lines. If your function is so long that you need a reminder for what function you're looking at, it's time to refactor it.